specs/2019-03/Cloning Branch Configuration/Cloning-Branch-Config-Spec.md

```diff
@@ -0,0 +1,38 @@
+# App Center Build Clone Branch Configuration Proposal
+
+This proposal details the approach the App Center team will take to address a heavily voiced frustration regarding branch configurations. We expect to start work on implementing this feature in the second quarter of 2019.
+
+## Problem Statement
+
+Currently, each time a new branch needs to be configured, one has to configure it completely from scratch. Existing configurations are not available for re-use. This is especially cumbersome for a branch configuration with many environment variables and when dealing with the management of signing credentials. Building a new branch via API today requires full configuration of the branch and uploading secrets.
+
+## Goal
+
+Given that at least one branch is already configured for a specific app, one can clone a configured branch’s configuration and apply it to a non-configured branch.
```

Comment:
It could be helpful also override a specific configuration. I believe that this will be available by default.

Reply:
This is a fair point. 👍 With Azure DevOps backing our build service this might be a bit tricky for us when it comes to the way we manage secrets. Cloning a configuration is an API call, but overwriting a config with new secrets without losing build logs is a more difficult challenge. @nrajpurkar and @DennisPan will take a look and circle back.

Reply:
While this is an obvious improvement over the current implementation I don't understand the limitation of one configuration per branch. Why isn't the branch a property of the configuration rather than the other way around? I would expect there to be an unlimited number of configurations with the ability to duplicate and delete any of them. I would choose the branch to be associated with each configuration just like I choose the version of Xcode, the scheme, etc. The way that configurations disappear in a cloud of smoke when I delete a release branch now is odd. I would like to have a release configuration that I use on any release branch as well as others. Some would run tests others wouldn't. The current design doesn't make much sense to me.

Reply:
@phoney this is definitely valid feedback and looks a little bit different than what is proposed here. That aligns more with #65 and also something we are taking a look into as a future improvement.